dev_tools/lib/solidus/release_drafter/builder.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+require 'set'
+
+module Solidus
+  class ReleaseDrafter
+    # Defines the template for releases and adds entries to it.
+    #
+    # @api private
+    class Builder
+      TITLE_PREFIX = '## '
+
+      TITLE_TEMPLATE = lambda do |category|
+        TITLE_PREFIX + category
+      end.freeze
```

Comment:
I'm curious to learn more about this pattern of implementing a small piece of functionality with a lambda. How does it differ from implementing the functionality with a class method (see below)?

```ruby
def self.title_template(category)
  TITLE_PREFIX + category
end
```

If you can point me to any docs or articles, I would appreciate it :)

Reply:
@gsmendoza, I can't name any specific pattern. I guess it's just going all the way down to treating a function just like any other value. In this case, we have other constants that are not functions that play a very similar role, so it felt more natural to me that all of them were constants. We could also argue that we could define everything as class methods, like in:

```ruby
def self.title_prefix
  '##'
end
```

The outcome would be exactly the same 🙂

Reply:
@waiting-for-dev I can't remember this rule of thumb I picked up before, but it's like "use the simplest tool for the problem at hand." So, for example, if you can implement a solution with both a function and a class, consider the function solution first, since functions are simpler than classes.

Using this rule of thumb, it makes more sense to implement `'##'` as a constant instead of as a class method, since a constant is simpler than a class method. 

A corollary to this idea is that, if you choose a more complex tool to implement a solution, it gives the impression that there was a reason behind it.

Going back to the example, if you choose to implement `'##'` as a class method, then readers of your code might assume that you might have had a reason behind that, since class methods are more complex than constants.

With lambdas versus class methods, I think that lambdas is the more powerful/fancier/esoteric of the two.

Your thoughts?

Reply:
Absolutely, @gsmendoza. Sorry if I wasn't clear enough, but I wasn't arguing to move those constants to class methods. I was using it to illustrate why it shouldn't be shocking to see lambdas as class methods, as they're another class of value, just like a string 🙂

Reply:
@waiting-for-dev I won't say "shocking". More like, "intriguing" :smile: